### PR TITLE
Changes to make health checking more reasonable

### PR DESCRIPTION
**Pull request by gabrieloczkowski**
Moved health checks to reasonable places; Using 'healthy' and 'unhealthy' instead of 'error' and 'ready' events;

### CODE REVIEW

File: src/lib/watchdog.coffee [hunk 1]
```diff
@@ -40,7 +40,11 @@ class Watchdog extends EventEmitter
   # Fetch tasks
   run: () ->
     async.forever (done) =>
-      return done new QueuerErrors.ShutdownError() if @isShuttingDown
+      if @isShuttingDown
+        @notifyHealthStatusChange 'unhealthy'
```
Comment (rlewan) [inline src/lib/watchdog.coffee]: `'healthy'` and `'unhealthy'` strings are used twice in this PR, perhaps it would be good to extract constants?

**Review by rlewan — APPROVED** [review]
👍 thanks, added just one small suggestion for improving this change set 🙂

**Review by tkopczuk — APPROVED** [review]
Approving to not hold it needlessly.

I'd reconsider using the Emitter pattern here, as in healthchecking business we're interested in the current state, not single events along the way. 

Perhaps a method / getter / variable `.health`.

I'd go for 'if polling didn't happen for some time, e.g `(3 * polling interval + job timeout)`, we're unhealthy, else we're healthy, combined with health coming from the underlying Mongo connection.

Seems a lot of code for something that should return 'healthy' if not shutting down, combined with additional code to handle this on the receiving side, to account for this being an emitter.

File: src/lib/watchdog.coffee (hunk 1 above)
Comment (gabrieloczkowski) [reply src/lib/watchdog.coffee]: 👍

**Comment by gabrieloczkowski** [comment]
Right now it is healthy when workers pooling new jobs and unhealthy when it is have problems with pooling or in shutdown-mode.
Created: https://swingdev.atlassian.net/browse/OPD-2041

**Pull request merged**